### PR TITLE
NumberBox: Restrict entering of a letter to editor with number mask (T1211093)

### PR DESCRIPTION
**Pull request by alexanderPolosatov**

### CODE REVIEW

File: packages/devextreme/js/ui/number_box/number_box.mask.js [hunk 1]
```diff
@@ -420,7 +420,11 @@ const NumberBoxMask = NumberBoxBase.inherit({
         const isValueChanged = parsedValue !== this._parsedValue;
 
         const isDecimalPointRestricted = char === decimalSeparator && maxPrecision === 0;
-        const isUselessCharRestricted = !isTextSelected && !isValueChanged && char !== MINUS && !this._isValueIncomplete(editedText) && this._isStub(char);
+        const isUselessCharRestricted =
+            !isTextSelected
```
Comment (ksercs) [inline packages/devextreme/js/ui/number_box/number_box.mask.js]: It's a kind of personal taste but don't u want to move `!isTextSelected` to the previous row? Naughty `=` looks strange to me :)

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 2]
```diff
@@ -1287,6 +1287,20 @@ QUnit.module('format: incomplete value', moduleConfig, () => {
         assert.equal(this.input.val(), '1.0', 'zero has not been removed');
     });
 
+    QUnit.test('NumberBox shold not accept letters in value when mas is "#0.0#" (T1211093)', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]:
```suggestion
    QUnit.test('NumberBox should not accept entering letters if mask is "#0.0#" (T1211093)', function(assert) {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 3]
```diff
@@ -1287,6 +1287,20 @@ QUnit.module('format: incomplete value', moduleConfig, () => {
...
+            format: '#0.0#',
+            value: '14.30',
+        });
+
+        this.keyboard
+            .caret(4)
+            .press('backspace')
+            .type('b');
+
+        assert.equal(this.input.val(), '14.0', 'value should not contain letter');
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]: Actually u check not a value here but input text
```suggestion
        assert.strictEqual(this.input.val(), '14.0', 'letter is not accepted');
```

File: packages/devextreme/js/ui/number_box/number_box.mask.js [hunk 4]
```diff
@@ -420,7 +420,11 @@ const NumberBoxMask = NumberBoxBase.inherit({
         const isValueChanged = parsedValue !== this._parsedValue;
 
         const isDecimalPointRestricted = char === decimalSeparator && maxPrecision === 0;
-        const isUselessCharRestricted = !isTextSelected && !isValueChanged && char !== MINUS && !this._isValueIncomplete(editedText) && this._isStub(char);
```
Comment (ksercs) [inline packages/devextreme/js/ui/number_box/number_box.mask.js:423]: It's hard to believe that this condition was added without a reason. Can we try to find a reason of its adding, maybe find a commit with this change?

Comment (alexanderPolosatov) [reply packages/devextreme/js/ui/number_box/number_box.mask.js:423]: checked. Looks what we can do this

File: packages/devextreme/js/ui/number_box/number_box.mask.js (hunk 1 above)
Comment (alexanderPolosatov) [reply packages/devextreme/js/ui/number_box/number_box.mask.js]: ok, thanks. fixed

**Review by marker-dao — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**